src/libraries/Microsoft.Extensions.Logging.Abstractions/src/Microsoft.Extensions.Logging.Abstractions.csproj

```diff
@@ -38,8 +38,10 @@ Microsoft.Extensions.Logging.Abstractions.NullLogger</PackageDescription>
 
   <ItemGroup>
     <AnalyzerReference Include="..\gen\Microsoft.Extensions.Logging.Generators.Roslyn3.11.csproj"
+                       Pack="true"
                        Condition="'$(DotNetBuildFromSource)' != 'true'" />
-    <AnalyzerReference Include="..\gen\Microsoft.Extensions.Logging.Generators.Roslyn4.0.csproj" />
+    <AnalyzerReference Include="..\gen\Microsoft.Extensions.Logging.Generators.Roslyn4.0.csproj"
```

Comment:
I'm not sure I like the overloading of `AnalyzerReference` to mean both:

* Include this analyzer in my NuGet package
* Reference this analyzer in my C# code

Here, the Microsoft.Extensions.Logging.Abstractions src library does not want the `Microsoft.Extensions.Logging.Generators` as an analyzer passed to the compiler. Even worse, we don't want 2 different versions of them passed at compile time. We just want them included in the NuGet tpackage.

Reply:
I just fixed that by introducing the `ReferenceAnalyzer` metadata on the item. As proposed via https://github.com/dotnet/runtime/issues/69111, the default behavior of an `AnalyzerReference` should be to just reference an analyzer project. By setting the `Pack` metadata, the analyzer also gets packaged, and by setting the `ReferenceAnalyzer` metadata, you can disable that the analyzer is referenced (but still gets packaged).